lib/feature/one_hot_encoder.cc

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "clink/feature/one_hot_encoder.h"
+
+#include <dirent.h>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <sys/stat.h>
+
+#include "nlohmann/json.hpp"
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
hmm.. why do we need `extern "C"` in this file? I thought we only need `extern "C"` in "*jna*" files?

Same for sparse_vector.cc.

Reply:
OK. I'll make the change.